java/src/org/openqa/selenium/chrome/ChromeDriver.java

```diff
@@ -96,10 +95,11 @@ private static ChromeDriverCommandExecutor generateExecutor(
     Require.nonNull("Driver service", service);
     Require.nonNull("Driver options", options);
     Require.nonNull("Driver clientConfig", clientConfig);
-    Result result = DriverFinder.getPath(service, options);
-    service.setExecutable(result.getDriverPath());
-    if (result.getBrowserPath() != null && !result.getBrowserPath().isEmpty()) {
-      options.setBinary(result.getBrowserPath());
+    DriverFinder finder = new DriverFinder(service, options);
+    service.setExecutable(finder.getDriverPath());
+    if (finder.hasBrowserPath()) {
+      options.setBinary(finder.getBrowserPath());
+      options.setCapability("browserVersion", (Object) null);
```

Comment:
But why do we want to remove the `browserVersion`?

Reply:
It's necessary in Firefox because Firefox does the compliant thing and fails if it doesn't match (e.g., "stable" will fail). Chrome it isn't technically necessary, but if Selenium Manager has matched the version and found the exact browser to use, sending the driver the version number is unnecessary.

Reply:
Like, per the spec, Chrome *should fail if the browser versions do not match, but it does not, but I figured the code should handle it in case it became spec compliant.

Reply:
Understood. We already do this in the Grid, and it was missing here. Thanks.

Reply:
It was being done in Driver Finder, this moves it up a level to the place that actually needs it.